addons/sale_coupon/tests/test_program_numbers.py

```diff
@@ -446,11 +447,17 @@ def test_program_numbers_one_discount_line_per_tax(self):
         # Free Pedal Bin       |  5  |   -100.00  | /        | -500.00 | -500.00 |       /
         # Free Large Cabinet   |  3  |   -100.00  | 15% excl | -300.00 | -345.00 |  -45.00
         # 20% on Large Cabinet |  1  |    -80.00  | 15% excl | -80.00  | -92.00  |  -12.00
+        # 10% on no tax        |  1  |    -50.00  | /        | -50.00  | -50.00  |       /
+        # 10% on tax 15% excl  |  1  |    -40.00  | 15% excl | -60.00  | -69.00  |   -9.00
```

Comment:
This line is misleading 🙂it's actually a 60$ price_unit.

Side note:
I compared the result with `sale_loyalty` here: https://github.com/odoo/odoo/blob/master/addons/sale_loyalty/tests/test_program_numbers.py#L554-L573

There is a slight change in behaviour so the amounts are slightly different but it seems correct to me.
You are applying -10% on 700+200-300 (free product) = 60$
Where as `sale_loyalty` is doing -10% on 400+200-80 (20% off) = 52$ (note that free products aren't represented as correction lines anymore)

```suggestion
        # 10% on tax 15% excl  |  1  |    -60.00  | 15% excl | -60.00  | -69.00  |   -9.00
```